### PR TITLE
Feat/historical query

### PR DESCRIPTION
**Pull request by yartman**
this pr adds from parameter to (GET /message) which enables retrieving message from certain point in time from a channel.

* plus adds multiple signatures and overloads for the subscribe api in the nats transport implementation.

### CODE REVIEW

File: apps/dsb-message-broker/src/message/message.query.pipe.ts [hunk 1]
```diff
@@ -3,8 +3,24 @@ import { PipeTransform, Injectable, BadRequestException } from '@nestjs/common';
 @Injectable()
 export class MessageQueryPipe implements PipeTransform<any> {
     async transform(data: any) {
-        if (!data.fqcn) throw new BadRequestException('fqcn is missing in the query!');
-        if (typeof data.fqcn !== 'string') throw new BadRequestException('fqcn must be a string!');
+        if (data.from) {
```
Comment (kosecki123) [inline apps/dsb-message-broker/src/message/message.query.pipe.ts:6]: We need to add better validation for incoming parameters, also I suggest we create a DTO object with class-validator annotations as we use in other DTO classes, there are plenty of built in validator for date, regex etc

File: apps/dsb-message-broker/src/message/message.query.pipe.ts [hunk 2]
```diff
@@ -3,8 +3,24 @@ import { PipeTransform, Injectable, BadRequestException } from '@nestjs/common';
...
+            if (!isNaN(data.from) || isNaN(Date.parse(data.from))) {
+                throw new BadRequestException(
+                    'The value of from parameter is not a valid dateTime.'
+                );
+            }
+
+            if (data.from) {
+                if (data.from.includes('.')) {
```
Comment (kosecki123) [inline apps/dsb-message-broker/src/message/message.query.pipe.ts:14]: Is that causes some sort of errors?

Comment (kosecki123) [reply apps/dsb-message-broker/src/message/message.query.pipe.ts:14]: ...by default Date().toISOString() returns date with ms, so we are asking users to make extra step, so I there a reason?

Comment (yartman) [reply apps/dsb-message-broker/src/message/message.query.pipe.ts:14]: in ISO format milliseconds are after dot and durable name for consumer cannot have dots. i fixed in next commit by replacing all dots with underscores.

File: apps/dsb-message-broker/src/message/message.query.pipe.ts (hunk 1 above)
Comment (yartman) [reply apps/dsb-message-broker/src/message/message.query.pipe.ts:6]: sure. creating it.

**Review by kosecki123 — APPROVED** [review]
👍

**Pull request merged**